### PR TITLE
Segmentate

### LINKED ISSUE #13: Segmentation Stage hinzufügen
@yvokeller @buesst1

### PR DESCRIPTION
**Pull request by yvokeller**

### CODE REVIEW

File: src/segmentate.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+print(
+    f"Segmentating {input_filename} to windows of {window_len_s}s with {overlap_percent}% overlap"
+)
+
+# read parquet file
+data = pd.read_parquet(input_filename)
+
+# helper function
+def segmentate(df:pd.DataFrame, window_len_s:float, overlap_percent:int):
+    """Makes windows [aka best os ;)] from dataframe
```
Comment (florinbarbisch) [inline src/segmentate.py:19]: true in all (edge) cases

File: src/segmentate.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        
+        window_start = window_end - overap_timedelta
+
+# segmentate
+window_dict = {}
+for group_name, dataframe in data.groupby(["activity", "person", "hash"]):
+    window_dict[group_name] = segmentate(dataframe, window_len_s, overlap_percent)
+
+# dump windows
+with open("data/segmentate.dill", "wb") as f:
```
Comment (florinbarbisch) [inline src/segmentate.py:55]: As of now, we don't want to segmentate uncalibrated data?

**Review by florinbarbisch — APPROVED** [review]
See second comment, for now this is fine to me.

**Comment by florinbarbisch** [comment]
Maybe use more meaningful commit messages in the future. 

https://dev.to/samuelfaure/how-atomic-git-commits-dramatically-increased-my-productivity-and-will-increase-yours-too-4a84

**Review by florinbarbisch — APPROVED** [review]

File: src/segmentate.py (hunk 2 above)
Comment (yvokeller) [reply src/segmentate.py:55]: we might want to, I don't see why this should not be possible though? there is no code specific to the sensor columns

Comment (florinbarbisch) [reply src/segmentate.py:55]: I think you want to solve this with a hyperparameter, which sounds like a good solution. But needs to be implemented first in db_pull.py. might open an issue later today

**Comment by gabrieltorresgamez** [comment]
can we merge this? @florinbarbisch

File: src/segmentate.py (hunk 1 above)
Comment (gabrieltorresgamez) [reply src/segmentate.py:19]: alternative fakten

**Comment by florinbarbisch** [comment]
Lol, will approve again:
image

**Review by florinbarbisch — APPROVED** [review]

**Pull request merged**